File: consentor-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__Consentor.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.attribute.ConsentAttributes;
import se.curity.identityserver.sdk.oauth.consent.ConsentorResult;
import se.curity.identityserver.sdk.oauth.consent.Consentor;

public final class ${pluginName}Consentor implements Consentor
{
    private final ${pluginName}ConsentorConfig _configuration;

    public ${pluginName}Consentor(${pluginName}ConsentorConfig configuration)
    {
        _configuration = configuration;

        // See examples of consentors on GitHub: https://github.com/search?q=topic%3Aconsentor+org%3Acurityio
    }

    @Override
    public ConsentorResult apply(ConsentAttributes consentAttributes, String transactionId)
    {
        throw new UnsupportedOperationException();
    }
}

File: kotlin-consentor-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__Consentor.kt
package ${package}.${pluginName}

import se.curity.identityserver.sdk.attribute.ConsentAttributes
import se.curity.identityserver.sdk.oauth.consent.ConsentorResult
import se.curity.identityserver.sdk.oauth.consent.Consentor

class ${pluginName}Consentor(private val _configuration: ${pluginName}ConsentorConfig): Consentor
{
    // See examples of consentors on GitHub: https://github.com/search?q=topic%3Aconsentor+org%3Acurityio

    override fun apply(consentAttributes: ConsentAttributes,transactionId: String): ConsentorResult
    {
        throw UnsupportedOperationException();
    }
}
